test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -241,23 +246,29 @@ database_operation::update_operation(thread_context *tc)
         }
 
         /* Start a transaction if possible. */
-        tc->transaction.try_begin(tc->session.get(), "");
+        tc->transaction.try_begin();
 
         /* Get the cursor associated with the collection. */
         scoped_cursor &cursor = cursors[coll.id];
 
         /* Choose a random key to update. */
         uint64_t key_id =
           random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
-        if (!tc->update(cursor, collection_id, tc->key_to_string(key_id)))
+        bool successful_update = tc->update(cursor, coll.id, tc->key_to_string(key_id));
+
+        /* Reset our cursor to avoid pinning content. */
+        testutil_check(cursor->reset(cursor.get()));
```

Comment:
Should the update call be responsible for this?

Reply:
I don't think so because if I was a user I wouldn't expect calling update to reset my position.

Reply:
Though the same could be said for rollbacks 🤔

Reply:
However `hs_cleanup` is a good example of why we don't want to do it.

Reply:
I mean the cursor is going to move when you call `update` anyway right? So it's not as if you can rely on it being in the same position as before you called `update`.

Reply:
I would expect it to be on the same key? Update should just update a value, no?